### PR TITLE
bring efel to c++17 standard

### PR DESCRIPTION
**Pull request by anilbey**
Changes:

* fixes linux, windows and mac wheel builds (previously failing)
* replace deprecated bind2nd with lambda
* add destructor to efel logger (to close the file once logger gets out of scope)

### CODE REVIEW

File: efel/cppcore/LibV1.cpp [hunk 1]
```diff
@@ -1006,13 +1003,13 @@ static int __time_constant(const vector<double>& v, const vector<double>& t,
...
-      return -1;
-    }
-    i_start++;
+                [min_derivative](double val) { return val > -min_derivative; }) !=
+        dvdt.begin() + i_start + 5) {
+      if (dvdt.begin() + i_start + 5 == dvdt.end()) {
+        GErrorStr += "Could not find the decay.\n";
+        return -1;
+      }
+      i_start++;
```
Comment (AurelienJaquier) [inline efel/cppcore/LibV1.cpp:1012]: why the indentation here?

File: efel/cppcore/LibV1.cpp [hunk 2]
```diff
@@ -1470,14 +1465,14 @@ static int __AP_width(const vector<double>& t, const vector<double>& v,
     v.begin() + indices[i + 1], bind2nd(less_equal<double>(), v_hm)));
     apwidth.push_back(t[hm_index2] - t[hm_index1]);
     */
-    int onset_index = distance(
-        v.begin(), find_if(v.begin() + indices[i], v.begin() + indices[i + 1],
-                           bind2nd(greater_equal<double>(), threshold)));
+    auto onset_index = std::distance(
+      v.begin(), std::find_if(v.begin() + indices[i], v.begin() + indices[i + 1],
```
Comment (AurelienJaquier) [inline efel/cppcore/LibV1.cpp:1469]: why use std:: here? You did not use it for the other changes

File: efel/cppcore/LibV1.cpp (hunk 1 above)
Comment (anilbey) [reply efel/cppcore/LibV1.cpp:1012]: I think I got confused with 4spaces vs 2 spaces indentation. Let me take it back

File: efel/cppcore/LibV1.cpp (hunk 2 above)
Comment (anilbey) [reply efel/cppcore/LibV1.cpp]: Done in the next commit. The usage is now consistent within LibV1

File: efel/cppcore/LibV5.cpp [hunk 3]
```diff
@@ -2400,20 +2398,20 @@ static int __sag_time_constant(const vector<double>& times,
...
   size_t min_length = 10;
 
-  // get start index
-  const size_t decayStartIdx =
-      distance(voltage.begin(),
-          find_if(voltage.begin(), voltage.end(),
-                  std::bind2nd(std::less_equal<double>(), minimum_voltage)));
+    // get start index
+    const size_t decayStartIdx =
+      std::distance(voltage.begin(),
```
Comment (AurelienJaquier) [inline efel/cppcore/LibV5.cpp:2403]: there are some more std::distance here

Comment (anilbey) [reply efel/cppcore/LibV5.cpp:2403]: The previous commit was for LibV1, the next ones repeat it for LibV5 for std::distance and std::find_if.

**Review by AurelienJaquier — APPROVED** [review]

**Pull request merged**